File: src/main/java/com/artillexstudios/axminions/integrations/Integrations.java
package com.artillexstudios.axminions.integrations;

import com.artillexstudios.axminions.integrations.implementation.block.BlockIntegration;
import com.artillexstudios.axminions.integrations.implementation.stacker.StackerIntegration;
import com.artillexstudios.axminions.integrations.implementation.storage.StorageIntegration;

public final class Integrations {
    public static final BlockIntegration BLOCK = new BlockIntegration();
    public static final StackerIntegration STACKER = new StackerIntegration();
    public static final StorageIntegration STORAGE = new StorageIntegration();
}

File: api/src/main/kotlin/com/artillexstudios/axminions/api/integrations/Integrations.kt
package com.artillexstudios.axminions.api.integrations

import com.artillexstudios.axminions.api.integrations.types.*

interface Integrations {

    fun getStackerIntegration(): StackerIntegration

    fun getPricesIntegration(): PricesIntegration?

    fun getEconomyIntegration(): EconomyIntegration?

    fun getIslandIntegration(): IslandIntegration?

    fun getProtectionIntegration(): ProtectionIntegrations

    fun reload()

    fun register(integration: Integration)

    fun deregister(integration: Integration)
}